orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/TitusInterestingHealthProviderNamesSupplier.java

```diff
@@ -0,0 +1,51 @@
+package com.netflix.spinnaker.orca.clouddriver.tasks.servergroup;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.spinnaker.orca.clouddriver.OortService;
+import com.netflix.spinnaker.orca.kato.pipeline.support.StageData;
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import retrofit.client.Response;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Arrays;
+
+@Slf4j
+@Component
+public class TitusInterestingHealthProviderNamesSupplier implements InterestingHealthProviderNamesSupplier {
+  private static final String TITUS = "Titus";
+  private static final String INTERESTING_HEALTH_PROVIDER_NAMES = "interestingHealthProviderNames";
+  private static final String ROLLING_PUSH = "rollingpush";
+  private static final List<String> SUPPORTED_STAGES = Arrays.asList("deploy", "enableServerGroup", "createServerGroup", "cloneServerGroup", "disableServerGroup");
+
+  @Override
+  public boolean supports(String cloudProvider, Stage stage) {
+    return cloudProvider.equalsIgnoreCase(TITUS) && SUPPORTED_STAGES.contains(stage.getType()) && stage.getContext().containsKey("strategy") && stage.getContext().get("strategy").equals(ROLLING_PUSH);
+  }
+
+  @Override
+  public List<String> process(String cloudProvider, OortService oortService, ObjectMapper objectMapper, Stage stage) {
+    String region = (String) stage.getContext().get("region");
+    String asgName = (String) stage.getContext().get("asgName");
+
+    try {
+      StageData stageData = (StageData) stage.mapTo(StageData.class);
+      Response response = oortService.getServerGroup(stageData.getApplication(), stageData.getAccount(), stageData.getCluster(), asgName, region, cloudProvider);
+      Map serverGroup = objectMapper.readValue(response.getBody().in(), Map.class);
+      Map titusServerGroupLabels = (Map) serverGroup.get("labels");
+
+      if (titusServerGroupLabels != null && titusServerGroupLabels.containsKey(INTERESTING_HEALTH_PROVIDER_NAMES)) {
+        String healthProviderNames = (String) titusServerGroupLabels.get(INTERESTING_HEALTH_PROVIDER_NAMES);
+        stage.getContext().put(INTERESTING_HEALTH_PROVIDER_NAMES, healthProviderNames);
```

Comment:
Should this method be modifying the stage context as a side effect?

Seems like the health provider names could be returned and dealt with by the caller?